### PR TITLE
feat: Added upgrade-python-requirements workflow

### PR DESCRIPTION
**Pull request by Jawayria**
JIRA: https://openedx.atlassian.net/browse/BOM-2597

### CODE REVIEW

**Review by mraarif — APPROVED** [review]

File: .github/workflows/upgrade-python-requirements.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          GITHUB_USER_EMAIL: ${{ secrets.REQUIREMENTS_BOT_GITHUB_EMAIL }}
+        run: |  
+          cd $GITHUB_WORKSPACE/testeng-ci
+          python -m jenkins.pull_request_creator --repo-root=$GITHUB_WORKSPACE \
+          --target-branch="${{ env.target_branch }}" --base-branch-name="upgrade-python-requirements" \
+          --commit-message="chore: Updating Python Requirements" --pr-title="Python Requirements Update" \
+          --pr-body="Python requirements update.Please review the [changelogs](https://openedx.atlassian.net/wiki/spaces/TE/pages/1001521320/Python+Package+Changelogs) for the upgraded packages." \
+          --user-reviewers="" --team-reviewers="community-engineering" --delete-old-pull-requests
+
+      - name: Send failure notification
```
Comment (sarina) [inline .github/workflows/upgrade-python-requirements.yml:60]: we've disabled this on our repos, thank you for removing this step. We get pinged using CODEOWNERS.

**Review by sarina — CHANGES_REQUESTED** [review]

File: .github/workflows/upgrade-python-requirements.yml (hunk 1 above)
Comment (sarina) [reply .github/workflows/upgrade-python-requirements.yml:60]: see https://github.com/edx/xblock-lti-consumer/blob/master/.github/workflows/upgrade-python-requirements.yml

File: .github/workflows/upgrade-python-requirements.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        if: ${{ failure() }}
+        uses: dawidd6/action-send-mail@v3
+        with:
+          server_address: email-smtp.us-east-1.amazonaws.com
+          server_port: 465
+          username: ${{secrets.EDX_SMTP_USERNAME}}
+          password: ${{secrets.EDX_SMTP_PASSWORD}}
+          subject: Upgrade python requirements workflow failed in ${{github.repository}}
+          to: aperture@edx.opsgenie.net
```
Comment (sarina) [inline .github/workflows/upgrade-python-requirements.yml:67]: this should be community-engineering, no?

Comment (Jawayria) [reply .github/workflows/upgrade-python-requirements.yml]: Yes updated

**Review by sarina — APPROVED** [review]

**Pull request merged**